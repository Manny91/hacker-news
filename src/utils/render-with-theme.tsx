import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "../styled-components";
import { theme } from "../theme";
import "jest-styled-components";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
