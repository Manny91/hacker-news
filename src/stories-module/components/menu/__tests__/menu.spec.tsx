import React from "react";
import Menu from "../menu";
import { renderWithTheme } from "../../../../utils/render-with-theme";

describe("Menu Component", () => {
  it("should render properly", () => {
    const subject = renderWithTheme(<Menu />);
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
