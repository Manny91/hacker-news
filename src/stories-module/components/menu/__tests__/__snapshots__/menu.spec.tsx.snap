// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component should render properly 1`] = `
.c3 {
  list-style-type: none;
  text-align: center;
  margin: 20px 0px 0px;
  cursor: pointer;
}

.c3.active {
  font-weight: bold;
}

.c3.active svg {
  color: #14137d;
}

.c5 {
  font-size: 10px;
  margin-top: 5px;
  color: #020202;
}

.c0 {
  height: 100%;
  border-right: 1px solid #f0f0f0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 20px 0px;
  height: 100%;
  margin: 0px;
}

.c8 {
  color: #4285f4;
  font-size: 26px;
}

.c4 {
  text-align: center;
  margin-bottom: 20px;
  cursor: pointer;
}

.c4 svg {
  height: 30px;
  width: 30px;
}

.c6 {
  color: #4285f4;
  font-size: 26px;
}

.c7 {
  color: #4285f4;
  font-size: 26px;
}

@media (max-width:992px) {
  .c3 {
    margin: 0px 20px 0px;
  }
}

@media (max-width:992px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (max-width:992px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div>
  <nav
    class="c0"
  >
    <ul
      class="c1"
    >
      <div
        class="c2"
      >
        <li
          class="c3"
        >
          <div
            class="c4"
          >
            <svg>
              logo.svg
            </svg>
          </div>
          <p
            class="c5"
          />
        </li>
        <li
          class="c3 active"
        >
          <span
            aria-label="align-left"
            class="anticon anticon-align-left c6"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="align-left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M120 230h496c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm0 424h496c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm784 140H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0-424H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
          <p
            class="c5"
          >
            Top Stories
          </p>
        </li>
        <li
          class="c3"
        >
          <span
            aria-label="area-chart"
            class="anticon anticon-area-chart c7"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="area-chart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-616-64h536c4.4 0 8-3.6 8-8V284c0-7.2-8.7-10.7-13.7-5.7L592 488.6l-125.4-124a8.03 8.03 0 00-11.3 0l-189 189.6a7.87 7.87 0 00-2.3 5.6V720c0 4.4 3.6 8 8 8z"
              />
            </svg>
          </span>
          <p
            class="c5"
          >
            Most Recent
          </p>
        </li>
      </div>
      <li
        class="c3"
      >
        <span
          aria-label="logout"
          class="anticon anticon-logout c8"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="logout"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 01-112.7 75.9A352.8 352.8 0 01512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 01-112.7-75.9 353.28 353.28 0 01-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
            />
          </svg>
        </span>
        <p
          class="c5"
        >
          Logout
        </p>
      </li>
    </ul>
  </nav>
</div>
`;
